lib/features/dragging/HoverFix.js

```diff
@@ -77,6 +77,16 @@ export default function HoverFix(eventBus, dragging, elementRegistry) {
 
   });
 
+  /**
+   * We make sure that dropping an element forces the element
+   * we're hovering over to lose the hover class.
+   */
+  eventBus.on('drag.end', function(event) {
+
+    canvas.removeMarker(event.hoverGfx, 'hover');
```

Comment:
I thought the issue was:

Generally `element.hover { A }` -> `element.out {A}` -> `element.hover {B}` is fired
Firefox sometimes swallows the `element.out {A}` so we end up with a sequence of  `element.hover { A }` -> `element.hover {B}` .

My intuition is that we should detect that broken sequence and emit the missing `element.out {A}`. If that is possible, it would be my preferred option.

Your solution drops the hover class, good. But it still leaves other components hanging that may rely on the proper sequence of `hover -> out -> hover` events.